src/librbd/ManagedLock.cc

```diff
@@ -109,15 +119,39 @@ void ManagedLock<I>::acquire_lock(Context *on_acquired) {
 }
 
 template <typename I>
-void ManagedLock<I>::try_acquire_lock(Context *on_acquired) {
+void ManagedLock<I>::try_acquire_exclusive_lock(Context *on_acquired) {
```

Comment:
Minor: instead of duplicating these acquire/try_acquire methods, I think they should either be collapsed into a single version that takes a enum for SHARED vs EXCLUSIVE (which would address my concerns about passing a bool around to indicate it is in EXCLUSIVE mode), or they should call a helper method that implements all the common logic.